### PR TITLE
codeintel: queue auto-indexing jobs for bulk-synced crates

### PR DESCRIPTION
**Pull request by Strum355**
Previously, we werent enqueueing autoindexing jobs for crate package repos bulk added on dotcom by the background syncer. There were no autoindexing policies that covered crates either, resulting in very few precise codeintel indexes for crate package repos.

This PR takes the approach of queueing jobs for everything on every sync. Ultimately only jobs will be queued for versions not already added (handled by enqueuer already), meaning we are tolerant to failures during the process of handling every crate

## Test plan

Ran locally _multiple_ times and backed in resilience to partial failures

### CODE REVIEW

File: internal/codeintel/dependencies/internal/background/job_cratesyncer.go [hunk 1]
```diff
@@ -167,14 +167,30 @@ func (b *crateSyncerJob) handleCrateSyncer(ctx context.Context, interval time.Du
...
+		} else if err != nil || externalService.LastSyncAt.After(nextSync) {
+			attemptsRemaining--
+			// mirrors backoff in job_dependency_indexing_scheduler.go
+			time.Sleep(time.Second * 30)
+			continue
+		}
+
+		break
+	}
+
```
Comment (efritz) [inline internal/codeintel/dependencies/internal/background/job_cratesyncer.go:193]: wot dis for

Comment (Strum355) [reply internal/codeintel/dependencies/internal/background/job_cratesyncer.go:193]: Its unfinished lol, need to wait for extsvc to have synced before i can enqueue jobs

Comment (efritz) [reply internal/codeintel/dependencies/internal/background/job_cratesyncer.go:193]: What happens here if we get through 5 attempts and don't hit the break condition?

What happens if we do/don't succeed here?

Comment (Strum355) [reply internal/codeintel/dependencies/internal/background/job_cratesyncer.go:193]: We try again in 12/configured interval time

File: internal/codeintel/dependencies/internal/background/job_cratesyncer.go [hunk 2]
```diff
@@ -167,14 +167,30 @@ func (b *crateSyncerJob) handleCrateSyncer(ctx context.Context, interval time.Du
...
 			return err
 		}
 	}
+
+	attemptsRemaining := 5
+	for {
+		externalService, err = b.extSvcStore.GetByID(ctx, externalService.ID)
+		if err != nil && attemptsRemaining == 0 {
+			return err
+		} else if err != nil || externalService.LastSyncAt.After(nextSync) {
```
Comment (efritz) [inline internal/codeintel/dependencies/internal/background/job_cratesyncer.go:184]: Is there an error we can match that's non-infra level to retry on?

**Review by efritz — APPROVED** [review]

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.17% (-4.91 kb) 🔽 | -0.30% (-43.04 kb) 🔽 | -0.33% (-38.13 kb) 🔽 | -0.83% (-6) 🔽 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/nsc/crate-syncer-autoindex/compare-report.html;tab=live_object) for a full comparison between the commits c6c19fe and 1361440 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by efritz — APPROVED** [review]

**Pull request merged**